PopPUNK/assign.py

```diff
@@ -237,42 +257,46 @@ def assign_query(dbFuncs,
         joinDBs(ref_db, output, output,
                 {"threads": threads, "strand_preserved": strand_preserved})
         if model.type == 'lineage':
-            genomeNetwork[min(model.ranks)].save(output + "/" + os.path.basename(output) + '_graph.gt', fmt = 'gt')
+            save_network(genomeNetwork[min(model.ranks)], prefix = output, suffix = '_graph', use_gpu = gpu_graph)
             # Save sparse distance matrices and updated model
             model.outPrefix = os.path.basename(output)
             model.save()
         else:
-            genomeNetwork.save(output + "/" + os.path.basename(output) + '_graph.gt', fmt = 'gt')
-
-        # Update distance matrices with all calculated distances
-        if distances == None:
-            distanceFiles = ref_db + "/" + os.path.basename(ref_db) + ".dists"
-        else:
-            distanceFiles = distances
+            save_network(genomeNetwork, prefix = output, suffix = '_graph', use_gpu = gpu_graph)
 
-        refList, refList_copy, self, rrDistMat = readPickle(distanceFiles,
-                                                            enforce_self = True)
+        # Load the previous distances
+        refList_loaded, refList_copy, self, rrDistMat = \
+            readPickle(distances,
+                       enforce_self = True)
+        # This should now always be true, otherwise both qrDistMat and sparse matrix
+        # may need reordering
+        assert(refList_loaded == rNames)
 
         combined_seq, core_distMat, acc_distMat = \
-            update_distance_matrices(refList, rrDistMat,
-                                    queryList, qrDistMat,
-                                    qqDistMat, threads = threads)
-        assert combined_seq == refList + queryList
+            update_distance_matrices(rNames, rrDistMat,
+                                     qNames, qrDistMat,
+                                     qqDistMat, threads = threads)
+        assert combined_seq == rNames + qNames
 
         # Get full distance matrix and save
         complete_distMat = \
             np.hstack((pp_sketchlib.squareToLong(core_distMat, threads).reshape(-1, 1),
                        pp_sketchlib.squareToLong(acc_distMat, threads).reshape(-1, 1)))
         storePickle(combined_seq, combined_seq, True, complete_distMat, dists_out)
 
+        # Copy model if needed
+        if output != model.outPrefix:
+            model.outPrefix = output
+            model.save()
```

Comment:
It would be better to add this as a method, rather than directly modifying an attribute of the class. i.e. in `ClusterFit` something like:
```
def copy(prefix):
   self.outPrefix = prefix
   save()
```

Reply:
Added in acbeeea.